packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,372 @@
+  enabledNetworkMap: {
+    persist: true,
+    anonymous: true,
+  },
+};
+
+export class NetworkEnablementController extends BaseController<
+  typeof controllerName,
+  NetworkEnablementControllerState,
+  NetworkEnablementControllerMessenger
+> {
+  protected messagingSystem: NetworkEnablementControllerMessenger;
+
+  /**
+   * Creates a NetworkEnablementController instance.
+   *
+   * @param args - The arguments to this function.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
+   */
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: NetworkEnablementControllerMessenger;
+    state?: Partial<NetworkEnablementControllerState>;
+  }) {
+    const defaultState = getDefaultNetworkEnablementControllerState();
+
+    // Deep merge enabledNetworkMap
+    const mergedEnabledNetworkMap = {
+      eip155: {
+        ...defaultState.enabledNetworkMap.eip155,
+        ...state?.enabledNetworkMap?.eip155,
+      },
+      solana: {
+        ...defaultState.enabledNetworkMap.solana,
+        ...state?.enabledNetworkMap?.solana,
+      },
+    };
+
+    // Call the constructor of BaseControllerV2
+    super({
+      messenger,
+      metadata,
+      name: controllerName,
+      state: { enabledNetworkMap: mergedEnabledNetworkMap },
+    });
+
+    this.messagingSystem = messenger;
+
+    // Subscribe to network state changes
+    this.messagingSystem.subscribe(
+      'NetworkController:networkAdded',
+      (networkAdded: NetworkConfiguration) => {
+        this.onNetworkAdded(networkAdded);
+      },
+    );
+
+    // Subscribe to network removal events
+    this.messagingSystem.subscribe(
+      'NetworkController:networkRemoved',
+      (networkRemoved: NetworkConfiguration) => {
+        this.onNetworkRemoved(networkRemoved);
+      },
+    );
+  }
+
+  /**
+   * Handles network added events.
+   * This method is called when a network is added to the network controller.
+   * The network is automatically enabled when added.
+   *
+   * @param networkAdded - The network added event.
+   */
+  private onNetworkAdded(networkAdded: NetworkConfiguration) {
+    this.setEvmEnabledNetwork(networkAdded.chainId);
+  }
+
+  /**
+   * Handles network removed events.
+   * This method is called when a network is removed from the network controller.
+   * The network is automatically disabled when removed.
+   *
+   * @param networkRemoved - The network removed event.
+   */
+  private onNetworkRemoved(networkRemoved: NetworkConfiguration) {
+    this.disableNetwork(networkRemoved.chainId);
+  }
+
+  /**
+   * Sets the enabled network for an EVM chain.
+   *
+   * @param chainId - The chain ID of the network to enable.
+   */
+  setEvmEnabledNetwork(chainId: Hex) {
+    const caipChainId = toEvmCaipChainId(chainId);
+    const { namespace } = parseCaipChainId(caipChainId);
+
+    if (namespace === KnownCaipNamespace.Eip155) {
```

Comment:
Nit: To address `@typescript-eslint/no-unsafe-enum-comparison` in `multichain-network-controller`, we’re currently casting to string, but I’m not sure if that’s the most appropriate solution.
```suggestion
    if (namespace === (KnownCaipNamespace.Eip155 as string)) {